README.md

```diff
@@ -71,17 +108,39 @@ We can now directly execute our `Hello World` application on a real connected de
 $ cargo apk run
 ```
 
+If the crate includes a runnable binary as suggested above, you will likely be greeted by the following error:
+
+```console
+$ cargo apk run
+error: extra arguments to `rustc` can only be passed to one target, consider filtering
+the package by passing, e.g., `--lib` or `--bin NAME` to specify a single target
+Error: Command `cargo rustc --target aarch64-linux-android -- -L hello_world_android/target/cargo-apk-temp-extra-link-libraries` had a non-zero exit code.
+```
+
+To solve this, add `--lib` to the run invocation, like so:
+```console
+$ cargo apk run --lib
+```
+
+### 5. Inspect the output
+
+`ndk-glue` redirects stdout and stderr to Android `logcat`, including the `println!("Hello World")` by the example above. See [Logging and stdout](##Logging-and-stdout) below how to access it.
+
+## Rendering to the window
+
+Android native apps have no easy access to [Android's User Interface](https://developer.android.com/guide/topics/ui) functionality (bar [JNI](##jni) interop). Applications can instead draw pixels directly to the window using [`ANativeWindow_lock`](https://developer.android.com/ndk/reference/group/a-native-window#group___a_native_window_1ga0b0e3b7d442dee83e1a1b42e5b0caee6), or use a graphics API like OpenGL or Vulkan for high performance rendering.
```

Comment:
This should perhaps be expanded and added in a separate PR. I need `ANativeWindow_lock` and will probably PR a safe wrapper for it soon which we should reference instead, and this should be elaborated to at least mention `winit` for the activity lifecycle and event handling (we're in the https://github.com/rust-windowing organisation after all...). I'm sure there are some GUI crates available like flutter which (a quick-start guide?) we might reference here.

CC @dvc94ch

Reply:
flutter requires a different tool. don't see `cargo apk` in its current form being too helpful here.